panic: runtime error: index out of range [1] with length 0

**Describe the bug**
![image](https://user-images.githubusercontent.com/40487732/163909664-d6e54733-5b72-483c-8ebb-54e7e433ba5c.png)


**I have been running docker for several weeks, but it suddenly shut down today. The error message is as follows:**

```js

2022/04/19 02:01:08 /workspace/pkg/service/rtcservice.go:255
2022-04-19T02:01:08.945Z        ERROR   livekit.udp_mux v2@v2.1.20/udp_mux.go:270       could not write packet: io: read/write on closed pipe
github.com/pion/ice/v2.(*UDPMuxDefault).connWorker
        /go/pkg/mod/github.com/pion/ice/v2@v2.1.20/udp_mux.go:270
2022-04-19T02:01:09.023Z        ERROR   livekit.udp_mux v2@v2.1.20/udp_mux.go:270       could not write packet: io: read/write on closed pipe
github.com/pion/ice/v2.(*UDPMuxDefault).connWorker
        /go/pkg/mod/github.com/pion/ice/v2@v2.1.20/udp_mux.go:270
2022-04-19T02:01:09.137Z        ERROR   livekit.udp_mux v2@v2.1.20/udp_mux.go:270       could not write packet: io: read/write on closed pipe
github.com/pion/ice/v2.(*UDPMuxDefault).connWorker
        /go/pkg/mod/github.com/pion/ice/v2@v2.1.20/udp_mux.go:270
2022-04-19T02:01:10.561Z        INFO    livekit rtc/participant.go:1082 mediaTrack published    {"room": "bfeIZhFpqRNUPovUDcZiHaMXARi3R1QHzRavavpYQtX", "roomID": "RM_ryENeJ3SKp8s", "participant": "1513404477712060416", "pID": "PA_wDEVfGiirLy3", "kind": "video", "trackID": "TR_kaWSQD5m8gej", "rid": "q", "SSRC": 2046483120}
2022-04-19T02:01:10.571Z        INFO    livekit rtc/participant.go:1082 mediaTrack published    {"room": "bfeIZhFpqRNUPovUDcZiHaMXARi3R1QHzRavavpYQtX", "roomID": "RM_ryENeJ3SKp8s", "participant": "1513404477712060416", "pID": "PA_wDEVfGiirLy3", "kind": "video", "trackID": "TR_kaWSQD5m8gej", "rid": "h", "SSRC": 829099077}
2022-04-19T02:01:10.580Z        INFO    livekit rtc/participant.go:1082 mediaTrack published    {"room": "bfeIZhFpqRNUPovUDcZiHaMXARi3R1QHzRavavpYQtX", "roomID": "RM_ryENeJ3SKp8s", "participant": "1513404477712060416", "pID": "PA_wDEVfGiirLy3", "kind": "audio", "trackID": "TR_gVDnyXMCcjPh", "rid": "", "SSRC": 2229251543}
2022-04-19T02:01:10.889Z        INFO    livekit rtc/participant.go:1082 mediaTrack published    {"room": "bfeIZhFpqRNUPovUDcZiHaMXARi3R1QHzRavavpYQtX", "roomID": "RM_ryENeJ3SKp8s", "participant": "1513404477712060416", "pID": "PA_wDEVfGiirLy3", "kind": "video", "trackID": "TR_kaWSQD5m8gej", "rid": "f", "SSRC": 4273629132}
2022-04-19T02:01:11.688Z        ERROR   livekit.udp_mux v2@v2.1.20/udp_mux.go:270       could not write packet: io: read/write on closed pipe
github.com/pion/ice/v2.(*UDPMuxDefault).connWorker
        /go/pkg/mod/github.com/pion/ice/v2@v2.1.20/udp_mux.go:270
2022-04-19T02:01:11.963Z        ERROR   livekit.udp_mux v2@v2.1.20/udp_mux.go:270       could not write packet: io: read/write on closed pipe
github.com/pion/ice/v2.(*UDPMuxDefault).connWorker
        /go/pkg/mod/github.com/pion/ice/v2@v2.1.20/udp_mux.go:270
2022-04-19T02:01:12.323Z        INFO    livekit rtc/participant.go:1082 mediaTrack published    {"room": "bfeIZhFpqRNUPovUDcZiHaMXARi3R1QHzRavavpYQtX", "roomID": "RM_ryENeJ3SKp8s", "participant": "1513342831165657088", "pID": "PA_U77sncySqUmB", "kind": "audio", "trackID": "TR_exQfihWY6tT2", "rid": "", "SSRC": 626761333}
2022-04-19T02:01:12.565Z        INFO    livekit rtc/participant.go:1082 mediaTrack published    {"room": "bfeIZhFpqRNUPovUDcZiHaMXARi3R1QHzRavavpYQtX", "roomID": "RM_ryENeJ3SKp8s", "participant": "1513342831165657088", "pID": "PA_U77sncySqUmB", "kind": "video", "trackID": "TR_fTQsiGcFcgzp", "rid": "q", "SSRC": 3382145979}
2022-04-19T02:01:12.576Z        INFO    livekit rtc/participant.go:1082 mediaTrack published    {"room": "bfeIZhFpqRNUPovUDcZiHaMXARi3R1QHzRavavpYQtX", "roomID": "RM_ryENeJ3SKp8s", "participant": "1513342831165657088", "pID": "PA_U77sncySqUmB", "kind": "video", "trackID": "TR_fTQsiGcFcgzp", "rid": "h", "SSRC": 505433239}
2022-04-19T02:01:12.717Z        ERROR   livekit.udp_mux v2@v2.1.20/udp_mux.go:270       could not write packet: io: read/write on closed pipe
github.com/pion/ice/v2.(*UDPMuxDefault).connWorker
        /go/pkg/mod/github.com/pion/ice/v2@v2.1.20/udp_mux.go:270
2022-04-19T02:01:12.808Z        INFO    livekit rtc/participant.go:1082 mediaTrack published    {"room": "bfeIZhFpqRNUPovUDcZiHaMXARi3R1QHzRavavpYQtX", "roomID": "RM_ryENeJ3SKp8s", "participant": "1513342831165657088", "pID": "PA_U77sncySqUmB", "kind": "video", "trackID": "TR_fTQsiGcFcgzp", "rid": "f", "SSRC": 2254075540}
2022-04-19T02:01:15.717Z        ERROR   livekit.udp_mux v2@v2.1.20/udp_mux.go:270       could not write packet: io: read/write on closed pipe
github.com/pion/ice/v2.(*UDPMuxDefault).connWorker
        /go/pkg/mod/github.com/pion/ice/v2@v2.1.20/udp_mux.go:270
2022-04-19T02:01:31.005Z        INFO    livekit rtc/room.go:489 closing room    {"room": "uobReSSFcGzJZtGIc319SNnSYaQPmZfWmhnEOY72hc1", "roomID": "RM_fwzURMLN5eVf"}
2022-04-19T02:01:31.005Z        INFO    livekit logger/logger.go:26     deleting room state     {"room": "uobReSSFcGzJZtGIc319SNnSYaQPmZfWmhnEOY72hc1"}
2022-04-19T02:01:31.005Z        INFO    livekit logger/logger.go:26     room closed
2022-04-19T02:01:36.756Z        INFO    livekit utils/opsqueue.go:58    ops queue full  {"room": "bfeIZhFpqRNUPovUDcZiHaMXARi3R1QHzRavavpYQtX", "roomID": "RM_ryENeJ3SKp8s", "participant": "1513342831165657088", "pID": "PA_U77sncySqUmB", "trackID": "TR_fTQsiGcFcgzp", "layer": 2}
panic: runtime error: index out of range [1] with length 0

goroutine 961679 [running]:
encoding/binary.bigEndian.PutUint16(...)
        /usr/local/go/src/encoding/binary/binary.go:106
github.com/livekit/livekit-server/pkg/sfu/twcc.(*Responder).writeRunLengthChunk(...)
        /workspace/pkg/sfu/twcc/twcc.go:272
github.com/livekit/livekit-server/pkg/sfu/twcc.(*Responder).buildTransportCCPacket(0xc0009a7c80)
        /workspace/pkg/sfu/twcc/twcc.go:160 +0xbc9
github.com/livekit/livekit-server/pkg/sfu/twcc.(*Responder).Push(0xc0009a7c80, 0x2cde, 0x16e729c0e582bdad, 0x0)
        /workspace/pkg/sfu/twcc/twcc.go:80 +0x1d6
github.com/livekit/livekit-server/pkg/rtc.(*MediaTrack).AddReceiver.func2(0x37a0, 0xc006eaafb0, 0x1)
        /workspace/pkg/rtc/mediatrack.go:135 +0x27
github.com/livekit/livekit-server/pkg/sfu/buffer.(*Buffer).processHeaderExtensions.func1()
        /workspace/pkg/sfu/buffer/buffer.go:463 +0x30
github.com/livekit/livekit-server/pkg/utils.(*OpsQueue).process(0x37)
        /workspace/pkg/utils/opsqueue.go:65 +0x3b
created by github.com/livekit/livekit-server/pkg/utils.(*OpsQueue).Start
        /workspace/pkg/utils/opsqueue.go:33 +0x5b

```

--- a/pkg/sfu/twcc/twcc.go
+++ b/pkg/sfu/twcc/twcc.go
@@ -96,7 +96,9 @@ func (t *Responder) buildTransportCCPacket() rtcp.RawPacket {
 	sort.Slice(t.extInfo, func(i, j int) bool {
 		return t.extInfo[i].ExtTSN < t.extInfo[j].ExtTSN
 	})
-	tccPkts := make([]rtpExtInfo, 0, int(float64(len(t.extInfo))*1.2))
+	maxTccPktsLen := int(float64(len(t.extInfo)) * 1.2)
+	tccPkts := make([]rtpExtInfo, 0, maxTccPktsLen)
+	var consumedExtInfo int
 	for _, tccExtInfo := range t.extInfo {
 		if tccExtInfo.ExtTSN < t.lastExtSN {
 			continue
@@ -108,8 +110,16 @@ func (t *Responder) buildTransportCCPacket() rtcp.RawPacket {
 		}
 		t.lastExtSN = tccExtInfo.ExtTSN
 		tccPkts = append(tccPkts, tccExtInfo)
+		consumedExtInfo++
+		if len(tccPkts) >= maxTccPktsLen {
+			break
+		}
 	}
-	t.extInfo = t.extInfo[:0]
+	if consumedExtInfo == len(t.extInfo) {
+		t.extInfo = t.extInfo[:0]
+	} else {
+		t.extInfo = t.extInfo[consumedExtInfo:]
+	}
 
 	firstRecv := false
 	same := true

--- a/pkg/sfu/twcc/twcc_test.go
+++ b/pkg/sfu/twcc/twcc_test.go
@@ -344,3 +344,15 @@ func BenchmarkBuildPacket(b *testing.B) {
 		_ = twcc.buildTransportCCPacket()
 	}
 }
+
+func TestTccWithPacketLost(t *testing.T) {
+	twcc := NewTransportWideCCResponder(123)
+	var fbreceived int
+	twcc.OnFeedback(func(p rtcp.RawPacket) { fbreceived++ })
+
+	for i := 0; i < 200; i++ {
+		twcc.Push(10000+uint16(i*70), time.Now().UnixNano()+int64(i)*1e6, false)
+	}
+
+	assert.Greater(t, fbreceived, 0)
+}

